src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/update/actions/AllocateProtectedResourcesAction.java

```diff
@@ -121,4 +122,17 @@ protected void onFailure(FlowUpdateFsm stateMachine) {
     protected String getGenericErrorMessage() {
         return "Could not update flow";
     }
+
+    @Override
+    protected void handleError(FlowUpdateFsm stateMachine, Exception ex, ErrorType errorType, boolean logTraceback) {
+        super.handleError(stateMachine, ex, errorType, logTraceback);
+        notifyFsmListenersOnFailure(stateMachine, stateMachine.getErrorReason(), errorType);
+    }
+
+    @Override
+    protected void notifyFsmListenersOnFailure(FlowUpdateFsm stateMachine, String errorMessage, ErrorType errorType) {
+        // Notify about failed allocation.
+        stateMachine.getEventListeners().forEach(listener ->
```

Comment:
Please, use "notifyEventListenersOnError" method available in FlowProcessingWithEventSupportFsm class.

Reply:
Fixed.